test/integration/request-flow/resolve-localhost.test.js

```diff
@@ -0,0 +1,85 @@
+var expect = require('chai').expect,
+    sinon = require('sinon'),
+    server = require('../../fixtures/server');
+
+describe('request to *.localhost', function () {
+    var httpServer,
+        testrun,
+        responseBody,
+        port;
+
+    before(function (done) {
+        httpServer = server.createHTTPServer();
+
+        httpServer.on('/', function (req, res) {
+            res.writeHead(200);
+            responseBody = 'Postman';
+            res.end(responseBody);
+        });
+
+
+        httpServer.listen(0, function (err) {
+            if (err) { return done(err); }
+
+            port = httpServer.port;
+
+            this.run({
+                collection: {
+                    item: [{
+                        request: {
+                            url: 'localhost:' + port,
+                            method: 'POST'
+                        }
+                    }, {
+                        request: {
+                            url: 'subdomain.localhost:' + port,
+                            method: 'POST'
+                        }
+                    }]
+                }
+            }, function (err, result) {
+                testrun = result;
+                done(err);
+            });
+        }.bind(this));
+    });
+
+    after(function (done) {
+        httpServer.destroy(done);
+    });
+
+    it('should complete the run', function () {
+        expect(testrun).to.be.ok;
+        sinon.assert.calledOnce(testrun.start);
+        sinon.assert.calledOnce(testrun.done);
+        sinon.assert.calledTwice(testrun.request);
+        sinon.assert.calledTwice(testrun.response);
+        sinon.assert.calledWith(testrun.done.getCall(0), null);
+    });
+
+    it('should send correct request and response for localhost', function () {
+        var err = testrun.response.getCall(0).args[0],
+            response = testrun.response.getCall(0).args[2],
+            request = testrun.response.getCall(0).args[3];
+
+        sinon.assert.calledWith(testrun.request.getCall(0), null);
+        sinon.assert.calledWith(testrun.response.getCall(0), null);
+
+        expect(err).to.equal(null);
+        expect(request.url.toString()).to.eql('localhost:' + port);
+        expect(String(response.stream), responseBody);
```

Comment:
This is asserting nothing.

Assertions should not be done using variables. Must be explicit!
```suggestion
        expect(response.text()).to.equal('POSTMAN');
```